source/common/pool_impl.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <stack>
+#include <vector>
+
+#include "nighthawk/common/poolable.h"
+
+#include "common/common/assert.h"
+
+namespace Nighthawk {
+
+template <typename Poolable> class PoolImpl {
+public:
+  using poolDeletionDelegate = std::function<void(Poolable*)>;
```

Comment:
I think we should capitalize the "p" in "pool" here.

Reply:
done